content_script.js

```diff
@@ -24,27 +19,24 @@ $(function() {
 
   // 4:30PM -> 203000
   function getTimeString(time) {
-    time_string = (parseInt(time.replace(/[:APM]/g, "")) +
-                  (time.match(/P/) ? 1200 : 0) -
-                  (time.match(/12:\d\dPM/) ? 1200 : 0) -
-                  (utc_offset * 100)) * 100;
-
-    if (time_string < 100000) time_string = "0" + time_string;
-
-    return time_string;
+    time_string = time.substr(0, time.length - 2);
+    parts = time_string.split(':');
+    if (parts[0].length != 2) {
+      parts[0] = '0' + parts[0];
+    }
+    if (parts[1].length != 2) {
+      parts[1] = '0' + parts[1];
+    }
+    time_string = parts.join('') + '00';
+    if (time.match(/PM/) && parts[0] != 12) {
+      time_string = parseInt(time_string) + 120000;
```

Comment:
It be nice to keep `time_string` as a string and not make it sometimes a number. Also, give `parseInt` its optional second argument since it doesn't treat strings that start with `0` as base 10:

```
  time_string = '' + (parseInt(time_string, 10) + 120000);
}
return time_string
```